### ISSUE TITLE
GitRepo status Active but bundle is in "Processed" status

### ISSUE THREAD

**Issue by kkaempf** [rancher/fleet#1478]
### Issue description:
In Rancher UI, the GitRepo State is Active, but one of the bundles is in "Processed" status

### Business impact:
None

[SURE-6158]

### Repro steps:

Create GitRepo to install LH  using https://github.com/mrolmedo/bundledependency.git as git repository.
The GitRepo is intended to deploy LH but fails because of the Kubernetes version. The bundles changes to "Processed" status, but the GitRepo Status does not show that the bundle has not been deployed.

### Workaround:
Is workararound available and implemented? No
What is the workaround:

### Actual behavior:
GitRepo shows as Active even though there is a bundle that is not deployed

### Expected behavior:
The GitRepo should update its status if the bundle is in Processed status as it does when a bundle is in Modified or Error Applied status

**Comment by thsnielsen** [comment]
for the curious support engineer. whats the difference between icebox and backlog please.

**Comment by kkaempf** [comment]
@thsnielsen 
Backlog: planned to be worked on 'soon'
Icebox: Undecided, might be worked on 'later' or 'never'

**Comment by manno** [comment]
- [ ] document processed condition
- [ ] document gitrepo active condition (does not mean resources are ready)


After this, we can decide if this is a bug, or confusing terminology.

**Comment by 0xavi0** [comment]
Aside from the need of having a better documentation around what **Active** `GitRepo` and **Processed** `Bundle` means, I think this issue is no longer happening in Rancher 2.9.x

I tried to recreate in different ways, but the `GitRepo` was always showing the error state.

We added several changes to the `GitRepo` status in the last version of Fleet.

Now, when I try to install a helm chart asking for a `k8s` version that is not matching the actual one I get this:
![Image](https://github.com/user-attachments/assets/25dfc2cd-f592-4b5f-8381-e9d30072fb5f)
![Image](https://github.com/user-attachments/assets/882a093e-b8be-44fc-a5be-2865503451eb)

Can I kindly ask to try again with the latest Rancher to see if the issue is still repeatable on your side?
